tests/FSharp.Stats.Tests/GeneralisedLinearModels.fs

```diff
@@ -0,0 +1,1012 @@
+    
+    let internal generateBaseMatrixAndVector (yColumn:string) (colsToDrop:string list) (frame:Frame<int,string>) =
+        let vector = 
+            frame
+            |> Frame.getCol yColumn
+            |> Series.values
+            |> Vector.ofSeq
+        let matrix = 
+            dropCols frame (yColumn::colsToDrop)
+            |> Frame.toJaggedArray
+            |> Matrix.ofJaggedArray
+            |> testingWithOneCreation
+        matrix,vector
+
+    let internal checkIfInvIsPossible (matrix:Matrix<float>) =
+        matrix  
+        |> Matrix.exists (fun x -> 
+            FSharp.Stats.Ops.isNan(x) ||
+            FSharp.Stats.Ops.isPosInf(x)||
+            FSharp.Stats.Ops.isNegInf(x)
+            )
+        |> not
+
+
+[<Tests>]
+let linkerFunctions = 
+    testList "Test Linker functions for GLM" [
+        testCase "LogLinkFunction" <| fun () ->
+            
+            let linkExpected        = 
+                [|
+                    FSharp.Stats.Ops.inf 
+                    6.78897174
+                    0.
+                    -36.04365339
+                    -36.04365339
+                    -36.04365339
+                    -36.04365339  
+                |]
+
+            let linkInvExpected        = 
+                [|
+                    FSharp.Stats.Ops.inf 
+                    FSharp.Stats.Ops.inf
+                    2.71828183
+                    1.
+                    0.36787944
+                    0.
+                    0.        
+                |]
+
+            let linkDerExpected  =
+                [|
+                    0.00000000e+00
+                    1.12612613e-03
+                    1.00000000e+00
+                    4.50359963e+15
+                    4.50359963e+15
+                    4.50359963e+15
+                    4.50359963e+15      
+                |]
+
+            let linkInvDerExpected  =
+                [|
+                    FSharp.Stats.Ops.inf
+                    FSharp.Stats.Ops.inf
+                    2.71828183
+                    1.
+                    0.36787944
+                    0.
+                    0.        
+                |]
+            
+            let link            = Fitting.GLM.LinkFunctions.LogLinkFunction 
+            let linkF           = link.getLink
+            let linkFInv        = link.getInvLink
+            let linkDer         = link.getDeriv
+            let linkFInvDer     = link.getInvLinkDerivative
+
+            let linkFActual             = testingArray |> Array.map(linkF) 
+            let linkFInvActual          = testingArray |> Array.map(linkFInv) 
+            let linkFDerActual          = testingArray |> Array.map(linkDer) 
+            let linkFInvDerActual       = testingArray |> Array.map(linkFInvDer) 
+
+            for i=0 to testingArray.Length-1 do
+                let expected    = linkExpected.[i]
+                let actual      = linkFActual.[i] 
+                if isInf actual then
```

Comment:
I do not understand this test setup. I think you should base the first conditional on the expected value and not on the actual value, e.g. "If the expected value is infinity then actual should be infinity", same for the tests below